zipline/data/data_portal.py

```diff
@@ -0,0 +1,1186 @@
+                    self._augmented_sources_map[col_name] = {}
+
+                self._augmented_sources_map[col_name][identifier] = df
+
+    def _open_daily_file(self):
+        if self._daily_equities_data is None:
+            self._daily_equities_data = bcolz.open(self._daily_equities_path)
+            self.daily_equities_attrs = self._daily_equities_data.attrs
+
+        return self._daily_equities_data, self.daily_equities_attrs
+
+    def _open_minute_file(self, field, sid):
+        if self._sid_path_func is None:
+            path = "{0}/{1}.bcolz".format(self._minutes_equities_path, sid)
+        else:
+            path = self._sid_path_func(self._minutes_equities_path, sid)
+
+        try:
+            carray = self._carrays[field][path]
+        except KeyError:
+            carray = self._carrays[field][path] = bcolz.carray(
+                rootdir=path + "/" + field, mode='r')
+
+        return carray
+
+    def get_previous_value(self, asset, field, dt):
+        """
+        Given an asset and a column and a dt, returns the previous value for
+        the same asset/column pair.  If this data portal is in minute mode,
+        it's the previous minute value, otherwise it's the previous day's
+        value.
+
+        Parameters
+        ---------
+        asset : Asset
+            The asset whose data is desired.
+
+        field: string
+            The desired field of the asset.  Valid values are "open",
+            "open_price", "high", "low", "close", "close_price", "volume", and
+            "price".
+
+        dt: pd.Timestamp
+            The timestamp from which to go back in time one slot.
+
+        Returns
+        -------
+        The value of the desired field at the desired time.
+        """
+        if self._data_frequency == 'daily':
+            prev_dt = self.env.previous_trading_day(dt)
+        elif self._data_frequency == 'minute':
+            prev_dt = self.env.previous_market_minute(dt)
+
+        return self.get_spot_value(asset, field, prev_dt)
+
+    def _check_fetcher(self, asset, column, day):
+        # if there is a fetcher column called "price", only look at it if
+        # it's on something like palladium and not AAPL (since our own price
+        # data always wins when dealing with assets)
+        look_in_augmented_sources = column in self._augmented_sources_map and \
+            not (column in BASE_FIELDS and isinstance(asset, Asset))
+
+        if look_in_augmented_sources:
+            # we're being asked for a fetcher field
+            try:
+                return self._augmented_sources_map[column][asset].\
+                    loc[day, column]
+            except:
+                log.error(
+                    "Could not find value for asset={0}, current_day={1},"
+                    "column={2}".format(
+                        str(asset),
+                        str(self.current_day),
+                        str(column)))
+
+                raise KeyError
```

Comment:
Could this KeyError contain more information for code that catches it?  Maybe move the logging into the error.